src/app/src/test/kotlin/org/vaccineimpact/orderlyweb/tests/unit_tests/controllers/web/WorkflowRunControllerTests.kt

```diff
@@ -316,6 +334,41 @@ class WorkflowRunControllerTests
             on { params(":key") } doReturn "workflow_key1"
             on { userProfile } doReturn CommonProfile().apply { id = "test@user.com" }
         }
+
+        val workflowRun = WorkflowRun(
+            "workflow",
+            "workflow_key1",
+            "test@user.com",
+            Instant.now(),
+            listOf(
+                WorkflowReportWithParams("Report A", emptyMap()),
```

Comment:
It would be great to test the bug case, where statuses were incorrectly matched to report names when returned in  different order to those in the details list, so could mix them up either here or in the mock API response and check the expected response is still returned. 

Also - it wasn't added in this branch but `mockAPIResponseText` below is misleading because it includes the report names (the fact that names aren't included in this response is what caused this issue in the first place!) - so while you're in this code it would be good to remove those, just to keep the test more realistic

Reply:
> It would be great to test the bug case, where statuses were incorrectly matched to report names when returned in different order to those in the details list, so could mix them up either here or in the mock API response and check the expected response is still returned.

I think this is a must - considering it was actually the motivation for this ticket